### PR TITLE
Detect negated content absent buffer 2224 v9

### PR DESCRIPTION
**Pull request by catenacyber**
Link to [redmine](https://redmine.openinfosecfoundation.org/projects/suricata/issues) ticket:
https://redmine.openinfosecfoundation.org/issues/2224

Describe changes:
- detect: negated content matches on absent buffer
- detect: adds `absent` keyword to match on absent buffer
- doc: fix `byte_test`examples

```
SV_BRANCH=pr/1535
```
https://github.com/OISF/suricata-verify/pull/1535

#10033 fixing debug validation by wrapping with check that `Callback == DetectEngineInspectBufferGeneric

### CODE REVIEW

**Comment by suricata-qa** [comment]
Information: QA ran without warnings. 

Pipeline 17015

**Comment by victorjulien** [comment]
Can you add a ticket for the doc fixes? Want to backport those.

File: src/detect.h [hunk 1]
```diff
@@ -424,6 +424,8 @@ typedef struct DetectEngineAppInspectionEngine_ {
     uint8_t id;     /**< per sig id used in state keeping */
     bool mpm;
     bool stream;
+    // will match on a NULL buffer (only negated content)
```
Comment (victorjulien) [inline src/detect.h:427]: can you use doxygen style comments for structs

**Comment by catenacyber** [comment]
> Can you add a ticket for the doc fixes? Want to backport those.

https://redmine.openinfosecfoundation.org/issues/6629

File: src/detect.h (hunk 1 above)
Comment (catenacyber) [reply src/detect.h:427]: /** ok  */

File: src/detect-engine.c [hunk 2]
```diff
@@ -696,6 +699,75 @@ static void AppendAppInspectEngine(DetectEngineCtx *de_ctx,
...
+                    SCLogDebug("sm->type %u", smd->type);
+                    DEBUG_VALIDATE_BUG_ON(1);
+            }
+            if (smd->is_last || !match_on_null) {
+                break;
+            }
+            // smd does not get reused after this loop
+            smd++;
+        }
+        new_engine->match_on_null = match_on_null;
```
[the remark this answers is not in the document]
Comment (catenacyber) [reply src/detect-engine.c:769]: Nice

**Comment by catenacyber** [comment]
Replaced by https://github.com/OISF/suricata/pull/10051

**Pull request closed without merging**